sdk/tools/azure-openrewrite/src/main/java/com/azure/openrewrite/core/http/HttpRequestCustomRecipe.java

```diff
@@ -0,0 +1,239 @@
+package com.azure.openrewrite.core.http;
+
+import com.azure.openrewrite.util.ConfiguredParserJavaTemplateBuilder;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.J;
+
+/**
+ * A custom OpenRewrite recipe to migrate the use of HttpRequest.
+ *
+ * <p>This recipe performs the following transformations:</p>
+ * <ul>
+ *   <li>Replaces the HttpRequest constructor with a fluent API.</li>
+ *   <li>Replaces the HttpRequest methods with clientcore HttpRequest methods.</li>
+ * </ul>
+ *
+ * <p>Example transformations:</p>
+ * <pre>
+ * Before: com.azure.core.http.HttpRequest &lt;constructor&gt;(com.azure.core.http.HttpMethod, java.lang.String)
+ * After: io.clientcore.core.http.models.HttpRequest &lt;constructor&gt;().setMethod(io.clientcore.core.http.models.HttpMethod).setUri(java.lang.String)
+ *
+ * Before: com.azure.core.http.HttpRequest setBody(java.lang.String)
+ * After: io.clientcore.core.http.models.HttpRequest setBody(io.clientcore.core.models.binarydata.BinaryData)
+ *
+ * Before: com.azure.core.http.HttpRequest getUrl()
+ * After: io.clientcore.core.http.models.HttpRequest getUri().toURL()
+ * </pre>
+ */
+public class HttpRequestCustomRecipe extends Recipe {
```

Comment:
Similar feedback as HttpHeadersCustomRecipe for maybeAddImport statment, javatemplate building